### PR TITLE
Display correct type for type hints

### LINKED ISSUE #1928: Type-hinted declaration's type is `Variant`
A declaration that uses a type hint instead of an explicit type, still has a known type:

    Dim startRow&

Selecting the above declaration displays `startRow (variable: Variant)` in the Rubberduck command bar; this is wrong. It should be the same as this:

    Dim startRow As Long

Which is displayed as `startRow (variable: Long)`.

Fixing this needs to be careful about the `AsTypeName` value of the declaration, since there is code that relies on it being empty for type-hinted declarations. Perhaps we should introduce a `KnownTypeName` property that takes the type hint into account, and use that in the command bar instead of the `AsTypeName`?

### PR DESCRIPTION
**Pull request by Hosch250**
Close #1928

### CODE REVIEW

**Comment by retailcoder** [comment]
The PR seems to ignore the discussion in #1928.. @Hosch250 what do you think of @Vogel612's input there?

**Comment by Hosch250** [comment]
I don't think we should be using the known type to display there.  That should be displayed as what the code declares it is.  I wouldn't be against adding a string "used as {0}" though.

File: Rubberduck.Parsing/Symbols/Declaration.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public class Declaration : IEquatable<Declaration>
                 "ANY"
         };
 
-        public static readonly IDictionary<string, string> TYPEHINT_TO_TYPENAME = new Dictionary<string, string>
```
Comment (retailcoder) [inline Rubberduck.Parsing/Symbols/Declaration.cs]: That dictionary should have been exposed as an `IReadOnlyDictionary<string,string>`, these mappings aren't meant to be tampered with. I'm still not convinced about exposing a public field here, when an instance property could do the job and simply expose a `HintedTypeName` and abstract away the data, the implementation detail that is this dictionary.

File: Rubberduck.Parsing/Symbols/Declaration.cs [hunk 2]
```diff
@@ -460,7 +460,7 @@ public bool AsTypeIsBaseType
         {
             get
             {
-                return string.IsNullOrWhiteSpace(AsTypeName) || BASE_TYPES.Contains(_asTypeName.ToUpperInvariant());
```
Comment (retailcoder) [inline Rubberduck.Parsing/Symbols/Declaration.cs]: The `BaseType` array is different. Well it should have been exposed as an `IEnumerable<string>`, but the point is that these values *have* to be `static`, and they gotta be *somewhere*. `Declaration` seems just convenient... for now. Eventually we'll need to implement a proper *ambient context*.

File: Rubberduck.Parsing/Symbols/Declaration.cs (hunk 1 above)
Comment (Hosch250) [inline Rubberduck.Parsing/Symbols/Declaration.cs]: I didn't add this, I just renamed it.

File: Rubberduck.Parsing/Symbols/Declaration.cs (hunk 2 above)
Comment (Hosch250) [inline Rubberduck.Parsing/Symbols/Declaration.cs]: Same here.

**Pull request merged**